popup.js

```diff
@@ -13,29 +14,63 @@ document.addEventListener('DOMContentLoaded', () => {
         // Load custom links if available
         if (result.customLinks) {
             result.customLinks.forEach(link => {
-                addCustomLink(link.name, link.url);
+                addCustomLink(link.name, link.url, false);
             });
         }
     });
 
-    // Save links
-    document.getElementById('save').addEventListener('click', () => {
+    // Load auto-save preference
+    chrome.storage.sync.get(['autoSave'], (result) => {
+        const isAutoSaveEnabled = result.autoSave !== undefined ? result.autoSave : true;
+        updateAutoSaveButton(isAutoSaveEnabled);
+    });
+
+    autoSaveButton.addEventListener('click', toggleAutoSave);
+
+    function toggleAutoSave() {
+        const isCurrentlyEnabled = autoSaveButton.classList.contains('auto-save-enabled');
+        const newAutoSaveState = !isCurrentlyEnabled;
+        chrome.storage.sync.set({ autoSave: newAutoSaveState });
+        updateAutoSaveButton(newAutoSaveState);
+    }
+
+    function updateAutoSaveButton(isEnabled) {
+        if (isEnabled) {
+            autoSaveButton.classList.add('auto-save-enabled');
+            autoSaveButton.classList.remove('auto-save-disabled');
+        } else {
+            autoSaveButton.classList.remove('auto-save-enabled');
+            autoSaveButton.classList.add('auto-save-disabled');
+        }
+    }
+
+    // Function to save all links
+    function saveAllLinks() {
```

Comment:
Since we have auto-save option, consider if auto-safe true, then disable the manual save button. If auto-safe false, then enable the manual save button.